packages/common/addon/services/web3.js

```diff
@@ -0,0 +1,17 @@
+import Service from '@ember/service';
+import Web3 from 'web3';
+
+export default Service.extend({
+  isMetamask: false,
+
+  init() {
+    this._super(arguments);
+
+    if (window.ethereum && window.ethereum.isMetaMask) {
+      let web3 = new Web3(window.ethereum);
+
+      this.set('eth', web3.eth);
+      this.set('isMetamask', true);
+    }
```

Comment:
You might wanna take a look at what I added here for tally to deal with the situation where the user changes the metamask account or login/logout https://github.com/cardstack/tally/blob/0c054d04e3cf6fa5db9a06e2e98c97b63480ee3e/packages/common/addon/services/web3.js#L49-L52